otherlibs/stdune/src/io.ml

```diff
@@ -179,6 +179,7 @@ module Copyfile = struct
   let sendfile_with_fallback ?chmod ~src ~dst () =
     try sendfile ?chmod ~src ~dst ()
     with Unix.Unix_error (EINVAL, "sendfile", _) ->
+      Unix.unlink dst;
```

Comment:
What if this call fails?

The way we handle the sendfile fallback seems error prone in general. We should should move the fallback code directly into `Copyfile.sendfile` and only catch errors from the surrounding `sendfile` syscall.

This way we can resume the fallback with the existing open fd rather than deleting and re-opening.

Reply:
I just did that, let me know what you think.